packages/fxa-payments-server/src/store/types.tsx

```diff
@@ -77,11 +78,19 @@ export interface Customer {
   subscriptions: Array<CustomerSubscription>;
 }
 
-export interface FetchState<T, E = any> {
-  error: E | null;
-  loading: boolean;
-  result: T | null;
-}
+export type FetchStateUninitialized = {
+  error: null;
+  loading: false;
+  result: null;
+};
+export type FetchStateLoading = { error: null; loading: true; result: null };
+export type FetchStateError<E> = { error: E; loading: false; result: null };
+export type FetchStateLoaded<T> = { error: null; loading: false; result: T };
+export type FetchState<T, E = any> =
```

Comment:
Breaking `FetchState` up into distinct states turns out to very handy for type inference in components. For example: once we get through conditions for `if (state.loading)` and `if (state.error)`, then typescript can infer that we're narrowed down to a `FetchStateLoaded` type and that `state.result` should have a value.